### DISCUSSION TITLE
-j option for presets

### DISCUSSION THREAD

**Discussion by Zakaroo222** [encode2m4v/encode2m4v#4] [Q&A]
I've exported a custom preset .json from the handbrake GUI, but when trying to use it with the -j switch, it doesn't appear to be recognized? I'm also getting a bad substitution error. Super new to working in terminal, so it's very likely I've messed somethings up...I tried to mod the .sh to fit most of the conversion I'm doing (mkv tro mp4) and made the kill switch the default. Attached a copy of  the .sh as a pdf...like I said, super new so please let me know what other stuff I should share. I'm using MacOS 13.3(a) on a i5 Macbook pro
[encode2mp4 copy.pdf](https://github.com/encode2m4v/encode2m4v/files/10892313/encode2mp4.copy.pdf)

 zmorton@Backup-Lappie qi % ./encode2mp4.sh -j "qu_preset.json"
./encode2mp4.sh: line 1009: ${arg^^}: bad substitution

Welcome to encode2mp4 - the MP4 to M4V transcoder! 

SUBTITLE EXTENSION MENU
-----------------------
1) No Subtitles	 3) .en.srt	  5) other
2) .srt		 4) _en.srt
Select the subtitle extension(.srt): 1
 - SELECTED - 1 - No Subtitles

Do you want all files upconverted to 1080? (no): 
./encode2mp4.sh: line 1140: ${do_up_convert^^}: bad substitution
encode2mp4 SETTINGS SUMMARY
------------------------------

ROOT DIRECTORY: "./"

OUTPUT DIRECTORY: <same as source>

SELECTED: KILL (DELETE) INPUT FILES AFTER PROCESSING

INPUT FILE EXTENSION: ".mkv"

OUTPUT FILE EXTENSION: ".mp4"

SELECTED: NO SUBTITLES

Up-convert to 1080: FALSE

TRANSCODER PRESET: "HQ 1080p30 Surround"

HIT ENTER TO START (or <ctl>+c to cancel)

Encoding file number 1 of 6 total files to encode:

From directory: ./Series 07/

ENCODING FILE - "QI XL - s07e01 - Gardens.mkv" - TO "QI XL - s07e01 - Gardens.mp4"

To Output directory: ./Series 07/

**Comment by encode2m4v** [comment]
Hello,

Thanks for the question. It was a bit difficult to scan the .pdf (due to line wrapping etc) but thanks for including it to help diagnose the issue(s).

It looks like you commented out the setting of the command line arguments array 
   see   -- >  #cmd_line_args_array=( "$@" )
   Since you put the # in front it is not setting what is found as the command line arguments...
   
If you remove that # then it should process your -j argument 

You may need to put the full path to the .json  file as the second argument.

I would also set the following in the .sh
    In the defaults section - change
        input_file_extension_set="not set"   -- TO -->  input_file_extension_set="set"
        output_file_extension_set="not set"  -- TO -->  output_file_extension_set="set"
        
        add_subs_set="not set"               -- TO -->  add_subs_set="set"
        subtitle_extension_set="not set"     -- TO -->  subtitle_extension_set="set"
                   then add         -- --> add_subs="FALSE"
        default_up_convert=""                -- TO -->  default_up_convert="FALSE"
        up_convert_set="not set"             -- TO -->  up_convert_set="set"


Sorry it is difficult to see how to change things to make it work the way you wanted. I think a lot more detailed comments would probably help that! I will plan on adding those in the next update.

Please let me know how it goes.

Regards!
